core/src/main/java/io/micronaut/core/convert/DefaultConversionService.java

```diff
@@ -968,7 +968,7 @@ protected void registerDefaultConverters() {
 
     private SimpleDateFormat resolveFormat(ConversionContext context) {
         AnnotationMetadata annotationMetadata = context.getAnnotationMetadata();
-        Optional<String> format = annotationMetadata.stringValue(Format.class);
+        Optional<String> format = context.getFormat();
```

Comment:
I think we should take this opportunity to also review this default format. We should probably change it to whatever Date.toString returns. I looked into it and it seems like that format is `EEE MMM dd HH:mm:ss zzz yyyy`. We also need to support the RFC format here as well since `@Header` can bind to a `java.util.Date`. We should add tests for that

Reply:
Changed it as suggested, and added a test for the different Temporal classes

I added `OffsetTime`, `LocalTime` and `Instant` to `TimeConverterRegistrar`

There are still a couple we don't have converter for in there, but I'm not sure how pressing `Year` and `YearMonth` are 🤔

PTAL